### PR TITLE
Fix #1130

### LINKED ISSUE #1130: Unhandled error in Deferred (RobotsTxtMiddleware)
**Dev story**:
Let's say spider downloads all .zip files from http://habrahabr.ru/post/212029/ page
Url with .zip files looks like this: http://layer6.jenkins.tox.im/job/qt_gui_win32/lastSuccessfulBuild/artifact/qt/build/release/TOX-Qt-GUI.zip

It's a polite spider, so settings file contains:
`ROBOTSTXT_OBEY = True`

Middleware parses habrahabr.ru robots.txt file as well as 'external' robots.txt file from layer6.jenkins.tox.im. It's expected behaviour. 
But if request will be returned with error then the output would be:
```
2015-04-02 17:06:16+0300 [habrahabr] DEBUG: Gave up retrying <GET http://layer6.jenkins.tox.im/robots.txt> (failed 1 times): DNS lookup failed: address 'layer6.jenkins.tox.im' not found: [Errno 8] nodename nor servname provided, or not known.

2015-04-02 17:06:16+0300 [-] ERROR: Unhandled error in Deferred:
2015-04-02 17:06:16+0300 [-] Unhandled Error
	Traceback (most recent call last):
	Failure: twisted.internet.error.DNSLookupError: DNS lookup failed: address 'layer6.jenkins.tox.im' not found: [Errno 8] nodename nor servname provided, or not known.
```

### PR DESCRIPTION
**Pull request by torymur**
fixed unhandled error in deferred (RobotsTxtMiddleware)

### CODE REVIEW

**Comment by kmike** [comment]
The PR looks good, but there is an unnecessary file commited.

**Comment by torymur** [comment]
Oh, sorry for that! Fixed.

**Comment by pablohoffman** [comment]
I'm happy to merge this as it currently stands, but I want to give a chance to @torymur to add tests until tomorrow. (robots middleware has tests in `tests/test_downloadermiddleware_robotstxt.py`)

**Comment by kmike** [comment]
Pablo is happy to merge ==> merge it!

File: scrapy/contrib/downloadermiddleware/robotstxt.py [hunk 1]
```diff
@@ -49,8 +49,14 @@ def robot_parser(self, request, spider):
             )
             dfd = self.crawler.engine.download(robotsreq, spider)
             dfd.addCallback(self._parse_robots)
+            dfd.addErrback(self._logerror, robotsreq, spider)
         return self._parsers[netloc]
 
+    def _logerror(self, failure, request, spider):
+        if failure.type is not IgnoreRequest:
+            log.msg(format="Error downloading %%(request)s: %s" % failure.value,
```
Comment (pablohoffman) [inline scrapy/contrib/downloadermiddleware/robotstxt.py]: double percentage symbol is a typo @torymur ?

Comment (Curita) [inline scrapy/contrib/downloadermiddleware/robotstxt.py]: No, first '%' formats '%s' at the end of the string with failure.value, and reduces one of the '%' symbols in %%(request)s. Then %(request)s is formatted inside log.msg using the call arguments.

Comment (dangra) [inline scrapy/contrib/downloadermiddleware/robotstxt.py]: any reason to not use `%(failure)` and pass `failure=failure.value` keyword argument ?

Comment (Curita) [inline scrapy/contrib/downloadermiddleware/robotstxt.py]: Not that I know of, I rebased this change in the python logging PR as such: https://github.com/scrapy/scrapy/pull/1060/files#diff-0b875f758ae81b8ccd8fb0b0a80278b2R60

Comment (dangra) [inline scrapy/contrib/downloadermiddleware/robotstxt.py]: :+1:

Comment (torymur) [inline scrapy/contrib/downloadermiddleware/robotstxt.py]: Oh, notifications land into spam folder. Julia, thanks for the answer :)

**Pull request merged**